Extensions/read_more_now.js

```diff
@@ -1,211 +1,104 @@
 //* TITLE Read More Now **//
-//* VERSION 1.4.7 **//
+//* VERSION 2.0.0 **//
 //* DESCRIPTION Read Mores in your dash **//
-//* DETAILS This extension allows you to read 'Read More' posts without leaving your dash. Just click on the 'Read More Now!' button on posts and XKit will automatically load and display the post on your dashboard. **//
-//* DEVELOPER STUDIOXENIX **//
+//* DETAILS This extension changes &quot;Keep Reading&quot; links into collapsible elements for easy viewing. **//
+//* DEVELOPER New-XKit **//
 //* FRAME false **//
 //* BETA false **//
 
 XKit.extensions.read_more_now = new Object({
 
 	running: false,
-	button_caption: "Read More Now!",
-	apiKey: XKit.api_key,
-
-	preferences: {
-		auto_open: {
-			text: "Auto-open all Read More posts (not recommended)",
-			default: false,
-			value: false
-		},
-		inline_button: {
-			text: "Show a smaller, inline button for Read More Now",
-			default: false,
-			value: false
-		},
-		process_keep_reading: {
-			text: "Show Read More Now for Keep Reading links",
-			default: true,
-			value: true
-		}
-	},
+	cache: {},
 
 	run: function() {
 		this.running = true;
+		XKit.tools.init_css("read_more_now");
+		XKit.post_listener.add("read_more_now", this.find_links);
+		this.find_links();
+	},
 
-		if (XKit.interface.where().queue === true || XKit.interface.where().drafts === true) {
-			return;
-		}
-
-		XKit.post_listener.add("read_more_now", XKit.extensions.read_more_now.do);
-		XKit.extensions.read_more_now.do();
-		$(document).on('click', '.xkit-read-more-now', function() {
-			$(this).addClass("disabled");
-			$(this).html("Retrieving post...");
-			var m_url = $(this).attr('data-post-url');
-			var json_page_parts = m_url.replace(/https?:\/\//, "").split("/");
-			var blog_name = json_page_parts[0];
-			var post_id = json_page_parts[2];
-			var m_cont = $(this);
-
-			var api_url = "https://api.tumblr.com/v2/blog/" + blog_name + "/posts" + "?api_key=" + XKit.extensions.read_more_now.apiKey + "&id=" + post_id;
-
-			GM_xmlhttpRequest({
-				method: "GET",
-				url: api_url,
-				json: true,
-				onerror: function() {
-					$(m_cont).removeClass("disabled");
-					$(m_cont).html(XKit.extensions.read_more_now.button_caption);
-					XKit.extensions.read_more_now.show_failed();
-				},
-				onload: function(response) {
-					var data = JSON.parse(response.responseText);
-					var m_object = data.response;
+	find_links: function() {
```

Comment:
the fact that the function named `find_links` is also responsible for 1) fetching the posts and 2) adding those contents to the links is kind of confusing for me. I'm also not completely happy with the idea of fetching every read more post into a details element immediately—can't we do it lazily when you click to reveal it?

Reply:
i know the intention of this comment was to just have me rename the function, but it made me realise i could give the rest of the contained code an extension object `this` by calling a different extension function within the `.each`, so i hope you don't mind `process_link` as the name of the do-everything function ^_^;

as for the fetching everything immediately part, that'll be resolved when i address your last comment, where I'll revert the UI change and with it the fetch time